### PR TITLE
Footer Component Added.

### PR DESCRIPTION
**Pull request by dsbharathkumar**

### CODE REVIEW

File: src/_variables.scss [hunk 1]
```diff
@@ -5,8 +5,9 @@ $primary-grey-dark: #373f50;
 
 $primary-red: #fe696a;
 $primary-red-dark: #fe3638;
+$footer-colour: black;
```
Comment (akashsinha13) [inline src/_variables.scss:8]: colour should be color

File: src/app/footer/footer.component.html [hunk 2]
```diff
@@ -0,0 +1,68 @@
+<div class="footer bg-footer-lighter pt-5">
+    <div class="container">
+        <div class="row pb-2">
+            <div class="col-md-4 col-sm-6">
+                <div class="widget widget-links widget-light pb-2 mb-4">
+                    <h3 class="widget-title text-light">{{ shopDepartment }}</h3>
+                    <ul class="widget-list">
+                        <li class="widget-list-item" *ngFor= "let shopDepartment of shopDepartments; let i=index;" >
```
Comment (akashsinha13) [inline src/app/footer/footer.component.html:8]: add trackby

File: src/app/footer/footer.component.html [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+                            <a class="widget-list-link" id={{i}} href="#">{{ shopDepartment.name }}</a>
```
Comment (akashsinha13) [inline src/app/footer/footer.component.html:9]: do we need id?

File: src/app/footer/footer.component.html [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+            </div>
+        </div>    
+    </div>  
+</div>    
+<div class="pt-5 bg-footer-slighter">
+    <div class="container">
+        <div class="row pb-3">
+            <div class="col-md-3 col-sm-6 mb-4" *ngFor="let label of footerLabels;let i=index;">
+                <div class="d-flex">
+                    <i class= {{label.className}}  style="font-size: 2.40rem;"></i>
```
Comment (akashsinha13) [inline src/app/footer/footer.component.html:59]: do not use style in html. You can add class if you want.

File: src/app/footer/footer.component.scss [hunk 5]
```diff
@@ -0,0 +1,215 @@
+@import '../../variables' ;
+
+.footer {
+    flex-shrink: 0;
+    display: block;
+}
+
+.pt-5 {
+    padding-top: 3rem !important;
```
Comment (akashsinha13) [inline src/app/footer/footer.component.scss:9]: if padding is 3rem then why name is pt-5?

File: src/app/footer/footer.component.scss [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+}  
+
+.col-md-4 {
+    flex: 0 0 auto;
+    width: 33.33333333%;
```
Comment (akashsinha13) [inline src/app/footer/footer.component.scss:14]: if we use md-4, by default bootstrap will divide the width into 3 parts with 33.33% for each. Why do we need to add it again?

File: src/app/footer/footer.component.scss [hunk 7]
```diff
@@ -0,0 +1,215 @@
...
+    display: block;
+    color: rgba(255,255,255,.65);
+    font-size: .875rem;
+    font-weight: normal;
+    text-decoration: none;
+}
+
+.widget{
+    color: #f7f9fd;
+    margin: 0px 0px 24px;
```
Comment (akashsinha13) [inline src/app/footer/footer.component.scss:51]: Try to avoid using of px; use rem instead

File: src/app/footer/footer.component.scss [hunk 8]
```diff
@@ -0,0 +1,215 @@
...
+    list-style: none;
+    .widget-links{
+        display: inline-block;
+    }
+}   
+.pb-2 {
+   padding-bottom: 0.5rem !important;
+}
+.mb-4 {
+   margin-bottom: 1.5rem !important;
```
Comment (akashsinha13) [inline src/app/footer/footer.component.scss:62]: name is not matching with the value provided

File: src/app/footer/footer.component.scss [hunk 9]
```diff
@@ -0,0 +1,215 @@
...
+.text-primary{
+    color: #fe696a !important;
+}
+
+.ps-3 {
+    padding-left: 1rem !important;
+}
+
+.pb-3 {
+    padding-bottom: 1rem !important;
```
Comment (akashsinha13) [inline src/app/footer/footer.component.scss:102]: do not add important for each prop.

File: src/app/footer/footer.component.scss [hunk 10]
```diff
@@ -0,0 +1,215 @@
...
+
+// Apple and Google Apps
+
+.btn-market {
+    display: inline-block;
+    padding: 0.375rem 0.875rem;
+    padding-left: 2.8125rem;
+    border-radius: 0.375rem;
+    background-position: center left 0.75rem;
+    background-color: #2b3445;
```
Comment (akashsinha13) [inline src/app/footer/footer.component.scss:144]: define the color in variable file

File: src/app/footer/footer.component.ts [hunk 11]
```diff
@@ -7,6 +7,49 @@ import { Component, OnInit } from '@angular/core';
 })
 export class FooterComponent implements OnInit {
 
+  shopDepartment: string = "Shop departments";
+  shippingInfo: string = "Account & shipping info";
+  aboutUs: string = "About us";
```
Comment (akashsinha13) [inline src/app/footer/footer.component.ts:12]: these strings will always be static. You can hardcode these in html file. For dynamic prop, we use this approach

**Review by akashsinha13 — CHANGES_REQUESTED** [review]

**Comment by akashsinha13** [comment]
Remove package-lock file. This file should not be committed unless there is any change in the package.json file.

File: src/_variables.scss (hunk 1 above)
Comment (dsbharathkumar) [reply src/_variables.scss:8]: @akashsinha13, modified

File: src/app/footer/footer.component.html (hunk 2 above)
Comment (dsbharathkumar) [reply src/app/footer/footer.component.html:8]: @akashsinha13  Added trackBy;

File: src/app/footer/footer.component.html (hunk 3 above)
Comment (dsbharathkumar) [reply src/app/footer/footer.component.html:9]: @akashsinha13, For reference i kept but doesn't add any value to functionality.

File: src/app/footer/footer.component.html (hunk 4 above)
Comment (dsbharathkumar) [reply src/app/footer/footer.component.html:59]: @akashsinha13, yes we are handling that property in CSS file.

File: src/app/footer/footer.component.scss (hunk 5 above)
Comment (dsbharathkumar) [reply src/app/footer/footer.component.scss:9]: @akashsinha13, modified this at many places with proper number

File: src/app/footer/footer.component.scss (hunk 6 above)
Comment (dsbharathkumar) [reply src/app/footer/footer.component.scss:14]: @akashsinha13, yes , removed the code.

File: src/app/footer/footer.component.scss (hunk 7 above)
Comment (dsbharathkumar) [reply src/app/footer/footer.component.scss:51]: @akashsinha13, modified to rem instead of px.

File: src/app/footer/footer.component.scss (hunk 8 above)
Comment (dsbharathkumar) [reply src/app/footer/footer.component.scss:62]: @akashsinha13 , i have taken reference from  https://tailwindcss.com/docs/padding. we can replace it if it is not a good practise.
image

File: src/app/footer/footer.component.scss (hunk 9 above)
Comment (dsbharathkumar) [reply src/app/footer/footer.component.scss:102]: @akashsinha13, checked all and removed un-necessary things.

File: src/app/footer/footer.component.scss (hunk 10 above)
Comment (dsbharathkumar) [reply src/app/footer/footer.component.scss:144]: @akashsinha13, modified and taking it from variable.scss

File: src/app/footer/footer.component.ts (hunk 11 above)
Comment (dsbharathkumar) [reply src/app/footer/footer.component.ts:12]: @akashsinha13, Those strings are hardcoded.

**Pull request merged**